pkg/postgres/query/builder.go

```diff
@@ -52,70 +53,45 @@ func (qb *Builder) AddFilterRequest(request *filter.Request) error {
 			err               error
 			valueIsList       bool
 			conditionTemplate string
-			conditionParams   []any
 		)
 		valueIsList, _, err = checkFieldValueType(field)
 		if err != nil {
-			return fmt.Errorf("error checking filter field value type '%s': %w", field, err)
+			return nil, fmt.Errorf("error checking filter field value type '%s': %w", field, err)
 		}
-		conditionTemplate, conditionParams, err =
-			composeQuery(qb.querySettings.FilterFieldMapping, field, valueIsList)
+		conditionTemplate, err = composeQuery(qb.querySettings.FilterFieldMapping, field, valueIsList)
 		if err != nil {
-			return fmt.Errorf("error composing query from filter field %w", err)
+			return nil, fmt.Errorf("error composing query from filter field %w", err)
 		}
 		if index > 0 {
 			qb.query.WriteString(fmt.Sprintf(" %s", logicOperator))
 		}
-		qb.query.WriteString(generateConditionalQuery(conditionParams, conditionTemplate))
+		args = append(args, extractFieldValues(field.Value)...)
+		qb.query.WriteString(conditionTemplate)
 	}
-	return nil
+	return
 }
 
-// generateConditionalQuery generates filter conditions based on the provided filter field value,
-// field mappings, logic operators, and a flag indicating if these are the final request fields.
-// If the field value is not of the expected type or cannot be converted to a string, it skips that field.
-// The finalRequestFields flag determines whether to append the logic operator at the end of the query.
-func generateConditionalQuery(conditionParams []any, column string) string {
-	var (
-		queryString strings.Builder
-		conditions  []string
-	)
-
-	queryString.WriteString(fmt.Sprintf(" %s (", column))
-	for _, conditionParam := range conditionParams {
-		params, ok := conditionParam.([]any)
-		if !ok {
-			continue
-		}
-		for _, condition := range params {
-			conditions = append(conditions, fmt.Sprintf("'%s'", condition))
-		}
+func extractFieldValues(value any) []any {
+	if params, ok := value.([]any); ok {
+		return params
 	}
-	conditionalQuery := strings.Join(conditions, ", ")
-	queryString.WriteString(fmt.Sprintf("%s)", conditionalQuery))
-	return queryString.String()
+	return []any{value}
 }
 
-// AddSorting appends sorting conditions to the query builder based on the provided sorting request.
+// addSorting appends sorting conditions to the query builder based on the provided sorting request.
 // It constructs the ORDER BY clause using the specified sort column and direction.
-func (qb *Builder) AddSorting(sort *sorting.Request) error {
+func (qb *Builder) addSorting(sort *sorting.Request) ([]any, error) {
```

Comment:
`sort.SortColumn` is of type `string`, so the sortingArg will always be a single string
```suggestion
func (qb *Builder) addSorting(sort *sorting.Request) (string, error) {
```

It is preferable to keep the type safety as long as possible. You can just convert the returned string to `any` before appending it to the args in `Builder.Build`